src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java

```diff
@@ -0,0 +1,174 @@
+                                                          final Config config) {
+
+        String database = sourceConfig.getString(MongoSourceConfig.DATABASE_CONFIG);
+        String collection = sourceConfig.getString(MongoSourceConfig.COLLECTION_CONFIG);
+
+        Optional<List<Document>> pipeline = sourceConfig.getPipeline();
+        ChangeStreamIterable<Document> changeStream;
+        if (database.isEmpty()) {
+            changeStream = pipeline.map(mongoClient::watch).orElse(mongoClient.watch());
+        } else if (collection.isEmpty()) {
+            MongoDatabase db = mongoClient.getDatabase(database);
+            changeStream = pipeline.map(db::watch).orElse(db.watch());
+        } else {
+            MongoCollection<Document> coll = mongoClient.getDatabase(database).getCollection(collection);
+            changeStream = pipeline.map(coll::watch).orElse(coll.watch());
+        }
+
+        try (MongoChangeStreamCursor<ChangeStreamDocument<Document>> cursor = changeStream.cursor()) {
+            cursor.tryNext();
+        } catch (Exception e) {
+            getConfigByName(config, MongoSourceConfig.CONNECTION_URI_CONFIG).ifPresent(c ->
+                    c.addErrorMessage(format("Unable to create a change stream cursor: %s ", e.getMessage()))
+            );
+        }
+    }
+
+    public static void validateHasReadWritePermissions(final MongoClient mongoClient, final MongoSinkConfig sinkConfig,
+                                                       final Config config) {
+        for (final String topic : sinkConfig.getTopics()) {
+            MongoSinkTopicConfig mongoSinkTopicConfig = sinkConfig.getMongoSinkTopicConfig(topic);
+            MongoNamespace namespace = mongoSinkTopicConfig.getNamespace();
+
+            boolean hasReadWritePermissions = false;
+            try {
+                hasReadWritePermissions = !mongoClient.getDatabase(namespace.getDatabaseName())
+                        .runCommand(BsonDocument.parse(format("{rolesInfo: {role: 'readWrite', db: '%s' }}", namespace.getDatabaseName())),
+                                BsonDocument.class)
+                        .getArray("roles", new BsonArray())
+                        .isEmpty();
+            } catch (MongoCommandException e) {
+                if (e.getErrorCode() == 13) {
+                    // Not Authorized to run rolesInfo command.
+                    hasReadWritePermissions = true; // We can't check as the user doesn't have the perms
+                } else if (e.getErrorCode() == 8000 && e.getErrorMessage().contains("user is not allowed to do action")) {
+                    // Not Authorized on Atlas to run the command
+                    hasReadWritePermissions = true; // We can't check as the user doesn't have the perms
+                }
+            } catch (Exception e) {
+                // Ignore any security or other exceptions
```

Comment:
Other errors such as Auth security errors are caught separately and the hasReadWritePermissions flag is left as false.